### DISCUSSION TITLE
Running init on sub-models

### DISCUSSION THREAD

**Discussion by williamjchen** [charmbracelet/bubbletea#836] [Q&A]
If I'm running a sub-model, is there any way to get the init to run? For example, I want to run the focus and blink the textinput when I switch pages

**Comment by bashbunni** [comment]
Hey @williamjchen typically the `Init` of the main model will run on program start, other than that, you'll have to manually call the `Init` of that model or any submodels to have it run. I hope that helps! Let me know :)

**Reply by GlamorousCar** [reply]
Hi @bashbunni , I don't understand how to initialize correctly for a model if its instance is created inside the update method of the parent model. 

I'm doing something like this

```go
func (m parentScreenModel) Update(msg tea.Msg) (tea.Model, tea.Cmd) {
case 
...
switch m.state {
case inputsView:
... \\ 1 submodel, as a result of a certain logic, the state changes to the progressView
case progressView:

if m.init == false {
	newEmty := *NewtestModel()
        newEmty.Init()

	m.connection = newEmty
	m.init = true

}
m.connection.Update(msg)```

It doesn't work anyway.
